Add release workflow

dhaiducek opened the pull request:
- https://github.com/stolostron/backlog/issues/26085

dhaiducek commented:
This is largely taken from `clusteradm`'s workflow and will generate a draft release whenever a tag is pushed. I've updated the Policy Generator release instructions accordingly: https://docs.google.com/document/d/1puzltjSbIoNiNA9uMhWLOZGyYPj69ZGIhvnxTo0ihc4/edit#

mprahl commented on .github/release.yml:
```diff
@@ -0,0 +1,36 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Optional:
```suggestion
        uses: actions/checkout@v3
```

dhaiducek replied:
There is no `v3` for `checkout`: https://github.com/actions/checkout/releases

mprahl replied:
@dhaiducek there is, but v2 is also supported still:
https://github.com/actions/checkout/releases/tag/v3.1.0

mprahl commented on .github/release.yml:
```diff
@@ -0,0 +1,36 @@
+        uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
```
```suggestion
        uses: actions/setup-go@v3
        with:
          go-version-file: go.mod
```

dhaiducek replied:
Ah, missed this one--thanks!

mprahl commented on .github/release.yml:
```diff
@@ -0,0 +1,36 @@
+
+      - name: Build project
+        run: |
+          make build-binary
```
```suggestion
          make build-release
```

dhaiducek commented:
Sorry--the file was in the `.github` folder instead of `.github/workflows` and I moved it, updated the wrong file, and then deleted the updated file. 🤦 Should be good now. 🙂
